crates/pallet-domains/src/tests.rs

```diff
@@ -861,11 +862,14 @@ fn test_invalid_total_rewards_fraud_proof() {
     });
 }
 
+type FraudProofFor<T> =
+    FraudProof<BlockNumberFor<T>, <T as frame_system::Config>::Hash, <T as Config>::DomainHeader>;
```

Comment:
This seems to indicate `FraudProof` should have a single generic `Header` rather than 3 generics. Or this alias is incorrect because block numer and hash are not the same as header.

Reply:
The Number and Hash are Consensus ones while the Header is the Domain's header. I did not move the first two with Consensus header since there was no use for rest of header items except these two

Reply:
On `pallet-domains` on Consensus chain `BlockNumberFor<T>, <T as frame_system::Config>::Hash` are the actual block number and hash types of the consensus chain. We need them on Fraud proof since `BundleEquivocationProof` variant holds BundleHeader which internally holds ER which hash Consensus block number and Consensus Block hash fields.

> those types are not coming from the same T even though they happen to be the same in practice.

I'm confused with this statement, can you elaborate what you mean? I may have missed something here

Reply:
The core is correct, I was reading it wrong :man_facepalming: